seacrowd/sea_datasets/ucla_phonetic/ucla_phonetic.py

```diff
@@ -0,0 +1,158 @@
+  author={Li, Xinjian and Mortensen, David R and Metze, Florian and Black, Alan W},
+  booktitle={ICASSP 2021-2021 IEEE International Conference on Acoustics, Speech and Signal Processing (ICASSP)},
+  pages={6958--6962},
+  year={2021},
+  organization={IEEE}
+}
+"""
+
+_DATASETNAME = "ucla_phonetic"
+
+_DESCRIPTION = """\
+This dataset contains audio recordings and phonetic transcriptions of word utterances for various low-resource SEA languages.
+Each language has a directory of text and audio files, with the latter forming one data subset.
+The dataset is prepared from the online UCLA phonetic dataset, which contains 7000 utterances across 100 low-resource languages, phonetically aligned using various automatic approaches, and manually fixed for misalignments.
+"""
+
+_HOMEPAGE = "https://github.com/xinjli/ucla-phonetic-corpus?tab=readme-ov-file"
+
+_LANGUAGES = ["ace", "brv", "hil", "hni", "ilo", "khm", "mak", "mya", "pam"]
+
+_LICENSE = Licenses.CC_BY_NC_SA_4_0.value
+
+_LOCAL = False
+
+_DATA_URL = "https://github.com/xinjli/ucla-phonetic-corpus/releases/download/v1.0/data.tar.gz"
+
+_SUPPORTED_TASKS = [Tasks.SPEECH_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+def seacrowd_config_constructor(lang, schema, version):
+    if lang not in _LANGUAGES:
+        raise ValueError(f"Invalid lang {lang}")
+
+    if schema not in ["source", "seacrowd_sptext"]:
+        raise ValueError(f"Invalid schema: {schema}")
+
+    return SEACrowdConfig(
+        name=f"ucla_phonetic_{lang}_{schema}",
+        version=datasets.Version(version),
+        description=f"UCLA Phonetic {schema} for {lang}",
+        schema=schema,
+        subset_id=f"{lang}_{schema}",
+    )
+
+
+class UCLAPhoneticDataset(datasets.GeneratorBasedBuilder):
+    """This dataset contains audio recordings and phonetic transcriptions of word utterances for various low-resource SEA languages."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = (
+        [
+            SEACrowdConfig(
+                name="ucla_phonetic_source",
+                version=datasets.Version(_SOURCE_VERSION),
+                description="UCLA Phonetic source for ace",
+                schema="source",
+                subset_id="ace_source",
+            ),
+            SEACrowdConfig(
+                name="ucla_phonetic_seacrowd_sptext",
+                version=datasets.Version(_SOURCE_VERSION),
+                description="UCLA Phonetic seacrowd+sptext for ace",
+                schema="seacrowd_sptext",
+                subset_id="ace_seacrowd_sptext",
+            ),
+        ]
+        + [seacrowd_config_constructor(lang, "source", _SOURCE_VERSION) for lang in _LANGUAGES]
+        + [seacrowd_config_constructor(lang, "seacrowd_sptext", _SEACROWD_VERSION) for lang in _LANGUAGES]
+    )
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_ace_source"
```

Comment:
What is `ace` again? If it's part of the dataset name, maybe we should put it into `_DATASETNAME`

Reply:
@ljvmiranda921 I think `ace` refers to `_LANGUAGES[0]`.
Sorry for suddenly butting in, saw the incoming email and thought this is kind of obvious, but CMIIW.